src/node/rpc/member_frontend.h

```diff
@@ -719,8 +797,16 @@ namespace ccf
         if (member->status == MemberStatus::ACCEPTED)
         {
           member->status = MemberStatus::ACTIVE;
+          members->put(args.caller_id, *member);
+
+          // New active members are allocated a new recovery share
+          if (!node.split_ledger_secrets(args.tx))
```

Comment:
I don't think we ought to transition the member to active if we cannot give them a share. They effectively aren't, for the purposes of the constitution.

Reply:
Even if the member is transitioned to `ACTIVE` before we issue the share, the `Tx` will be aborted aborted altogether if `split_ledger_secrets()` fails and the member will not actually be recorded as `ACTIVE`?

Reply:
Yes, especially so I would think. The member would be active on paper and potentially in recovery quotas but unable to provide a share. That does not sound good.

Is there a scenario where we think there could be a transient failure to share that can be fixed with a re-share? Even if there is, transitioning the member without them having a share seems risky...

Reply:
As discussed, this is fine as the handler is transactional.